src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java

```diff
@@ -1233,6 +1260,14 @@ public PrivateChannel createPrivateChannel(DataObject json)
         if (channel != null)
             return channel;
 
+        //if we don't have enough information to construct the channel properly, the rest of the properties can be null
+        if (!json.hasKey("recipients") && !json.hasKey("recipient"))
+        {
+            PrivateChannelImpl priv = new PrivateChannelImpl(channelId, getJDA());
+            cachePrivateChannel(priv);
```

Comment:
This isn't playing back the cache

Reply:
I'd recommend adding the cache playback to run at the end of the `cachePrivateChannel` method